apps/smart-contracts/core/test/unit/PrePOMarket.test.ts

```diff
@@ -744,13 +768,37 @@ describe('=> prePOMarket', () => {
       const shortToRedeem = amountMinted
       const totalOwed = await calculateTotalOwed(longToRedeem, shortToRedeem, false)
 
-      await prePOMarket.connect(user).redeem(longToRedeem, shortToRedeem)
+      await prePOMarket.connect(user).redeem(longToRedeem, shortToRedeem, user.address)
 
       expect(await longToken.balanceOf(user.address)).to.eq(amountMinted.sub(longToRedeem))
       expect(await shortToken.balanceOf(user.address)).to.eq(amountMinted.sub(shortToRedeem))
       expect(await collateralToken.balanceOf(user.address)).to.eq(totalOwed)
     })
 
+    it('correctly settles redemption with recipient != redeemer after market end', async () => {
+      expect(user.address).not.eq(recipient.address)
+
+      const amountMinted = await setupMarketToEnd(TEST_FINAL_LONG_PAYOUT)
+      const longToRedeem = amountMinted
+      const shortToRedeem = amountMinted
+      const totalOwed = await calculateTotalOwed(longToRedeem, shortToRedeem, false)
+
+      await prePOMarket.connect(user).redeem(longToRedeem, shortToRedeem, recipient.address)
+
+      const filter = prePOMarket.filters.Redemption(user.address, recipient.address)
+      const events = await prePOMarket.queryFilter(filter)
+      const event = events[0].args
+
+      expect(event.redeemer).to.eq(user.address)
+      expect(event.recipient).to.eq(recipient.address)
+      expect(event.amountAfterFee).to.eq(totalOwed)
+      expect(event.fee).to.eq(0)
+
+      expect(await longToken.balanceOf(user.address)).to.eq(amountMinted.sub(longToRedeem))
+      expect(await shortToken.balanceOf(user.address)).to.eq(amountMinted.sub(shortToRedeem))
+      expect(await collateralToken.balanceOf(recipient.address)).to.eq(totalOwed)
```

Comment:
arrange section should always be grouped together
```suggestion
      const filter = prePOMarket.filters.Redemption(user.address, recipient.address)
      const events = await prePOMarket.queryFilter(filter)
      const event = events[0].args
      expect(event.redeemer).to.eq(user.address)
      expect(event.recipient).to.eq(recipient.address)
      expect(event.amountAfterFee).to.eq(totalOwed)
      expect(event.fee).to.eq(0)
      expect(await longToken.balanceOf(user.address)).to.eq(amountMinted.sub(longToRedeem))
      expect(await shortToken.balanceOf(user.address)).to.eq(amountMinted.sub(shortToRedeem))
      expect(await collateralToken.balanceOf(recipient.address)).to.eq(totalOwed)
```